src/ray/object_manager/pull_manager.cc

```diff
@@ -350,24 +355,37 @@ void PullManager::TryToMakeObjectLocal(const ObjectID &object_id) {
   // sending RPCs.
   if (!request.spilled_url.empty()) {
     const auto spilled_node_id = request.spilled_node_id;
+    // Set a flag on the request so we don't send duplicate restoration requests.
+    request.is_being_restored = true;
     restore_spilled_object_(
         object_id, request.spilled_url, spilled_node_id,
-        [this, object_id, spilled_node_id](const ray::Status &status) {
+        [this, object_id, &request, spilled_node_id](const ray::Status &status) {
```

Comment:
We may want to look up the pull request within the callback instead of capturing it by reference here, although I don't think that we're in danger of that captured reference getting invalidated?

Reply:
It could be dangerous during destruction, I think you should change it to lookup.

Reply:
I _think_ that the thread that runs this handler should always run in a thread that's joined by the thread that created the pull manager, so the handler should never run after the pull requests table has been destructed, and IIUC we aren't at risk of this particular pull request being deleted or the reference being invalidated by concurrent table mutation...

But, since this isn't a super critical path, I think that it'd be better to change it to a lookup for the sake of simplicity and guarding against future bugs. 👍